File: 6/list-6.py
class Motor:
    def __init__(self, modelo):
        self.modelo = modelo

class Carro:
    def __init__(self, marca, motor): 
        self.marca = marca
        self.motor = motor 

    def detalhes(self):
        return f"O carro da marca {self.marca} tem um motor {self.motor.modelo}"

def main():
    motor = Motor("V8")  
    carro = Carro("Volkswagen", motor) 

    print(carro.detalhes())  

if __name__ == "__main__":
    main()

File: 6/list-6.go
package main

import "fmt"

type Motor struct {
	modelo string
}

func (m Motor) Detalhes() string {
	return fmt.Sprintf("Modelo do motor: %s", m.modelo)
}

type Carro struct {
	marca string
	motor Motor
}

func (c Carro) Detalhes() string {
	return fmt.Sprintf("Marca: %s, %s", c.marca, c.motor.Detalhes())
}

func main() {
	motor := Motor{modelo: "v8"}
	carro := Carro{motor: motor, marca: "Fusca"}

	fmt.Println(carro.Detalhes())
}
